docs/README.rst

```diff
@@ -13,11 +13,45 @@ After each commit on ``master``, documentation is updated and published to `Read
 Build
 -----
 
+It is not necessary to re-build this documentation while modifying LightGBM's source code.
+The HTML files generated using ``Sphinx`` are not checked into source control.
+However, you may want to build them locally during development to test changes.
+
+Docker
+^^^^^^
+
+The most reliable way to build the documentation locally is with Docker, using `the same images readthedocs uses <https://hub.docker.com/r/readthedocs/build>`_.
+
+Run the following from the root of this repository to pull the relevant image and run a container locally.
+
+.. code:: sh
+
+    docker run \
+        --rm \
+        --user=0 \
+        -v $(pwd):/opt/LightGBM \
+        --env C_API=true \
+        --env CONDA=/opt/miniforge \
+        --env READTHEDOCS=true \
+        --workdir=/opt/LightGBM/docs \
+        --entrypoint="" \
+        -it readthedocs/build:ubuntu-20.04-2021.09.23 \
+        /bin/bash build-docs.sh
+
+When that code completes, open ``docs/html/index.html`` in your browser.
+
+.. note::
+
+    The navigation in locally-built docs does not link to the local copy of the R documentation. To view the local version of the R docs, open ``docs/_build/html/R/index.html`` in your browser.
```

Comment:
I think we have some confusing wordings in this doc about R docs.

There are two subsections `Docker` and `Locally` in this doc. TBH, I think that Docker can be treated as "building locally" actually. But it's just a nit. The problem is that in the `Docker` section we use `in locally-built docs` phrase which can be easily treated as a reference to the `Locally` section. Let me suggest to replace `Locally` with `Manually`.

Reply:
what do you think about just calling the other section `Without Docker`, to eliminate any confusion?

I tried that here: https://github.com/microsoft/LightGBM/pull/4957/commits/8010f99fda097fa9702b9c900f31d5d078eaf1fc

Reply:
I love it! 🙂